TestChargingCurrent.h

```diff
@@ -0,0 +1,65 @@
+#ifndef TESTCHARGINGCURRENT
+#define TESTCHARGINGCURRENT
+
+void testForEmptyCurrentSample(ChargingCurrent &chargingCurrent, std::vector<int> &currentSamples, std::stringstream &expectedOutput)
+{
+    currentSamples = {};
+    expectedOutput.str("");
+    expectedOutput << "\nNo Current Samples are given!!!";
+    assert(chargingCurrent.classifyChargingCurrentRange(currentSamples) == expectedOutput.str());
+}
+
+void testForOnePositiveCurrentSample(ChargingCurrent &chargingCurrent, std::vector<int> &currentSamples, std::stringstream &expectedOutput)
+{
+    currentSamples = {2};
+    expectedOutput.str("");
+    expectedOutput << "\nRange, Readings\n2-2, 1";
+    assert(chargingCurrent.classifyChargingCurrentRange(currentSamples) == expectedOutput.str());
+}
+
+void testForTwoPositiveCurrentSamples(ChargingCurrent &chargingCurrent, std::vector<int> &currentSamples, std::stringstream &expectedOutput)
+{
+    currentSamples = {4,5};
+    expectedOutput.str("");
+    expectedOutput << "\nRange, Readings\n4-5, 2";
+    assert(chargingCurrent.classifyChargingCurrentRange(currentSamples) == expectedOutput.str());
+}
+
+void testForPositiveCurrentSamples(ChargingCurrent &chargingCurrent, std::vector<int> &currentSamples, std::stringstream &expectedOutput)
+{
+    currentSamples = {10,12,9,11,16,1,3,3,1,4,2};
+    expectedOutput.str("");
+    expectedOutput << "\nRange, Readings\n1-4, 6\n9-12, 4\n16-16, 1";
+    assert(chargingCurrent.classifyChargingCurrentRange(currentSamples) == expectedOutput.str());
+}
+
+void testForNegativeCurrentSamples(ChargingCurrent &chargingCurrent, std::vector<int> &currentSamples, std::stringstream &expectedOutput)
+{
+    currentSamples = {-4,-5,-4,-3,-7,-8,-8,-3};

```

Comment:
Negative currents - good coverage of data-ranges and nice illustration of the output in the test. Such a test helps to discuss with stakeholders: is -8--7 understandable for a reader? When are negative currents possible (e.g., while driving the vehicle?)
No need to change the code here. Just wanted to mention that well-written tests like this will help us to communicate with stakeholders.